src/Servers/Kestrel/Core/src/Middleware/HttpsConnectionMiddleware.cs

```diff
@@ -299,4 +319,40 @@ private static X509Certificate2 ConvertToX509Certificate2(X509Certificate certif
             return new X509Certificate2(certificate);
         }
     }
+
+    internal static class HttpsConnectionMiddlewareLoggerExtensions
+    {
+
+        private static readonly Action<ILogger, Exception> _authenticationFailed =
+            LoggerMessage.Define(
+                logLevel: LogLevel.Debug,
+                eventId: new EventId(1, "AuthenticationFailed"),
+                formatString: CoreStrings.AuthenticationFailed);
+
+        private static readonly Action<ILogger, Exception> _authenticationTimedOut =
+            LoggerMessage.Define(
+                logLevel: LogLevel.Debug,
+                eventId: new EventId(2, "AuthenticationTimedOut"),
+                formatString: CoreStrings.AuthenticationTimedOut);
+
+        private static readonly Action<ILogger, string, SslProtocols, Exception> _httpsConnectionEstablished =
+            LoggerMessage.Define<string, SslProtocols>(
+                logLevel: LogLevel.Debug,
+                eventId: new EventId(3, "HttpsConnectionEstablished"),
+                formatString: CoreStrings.HttpsConnectionEstablished);
+
+        private static readonly Action<ILogger, Exception> _http2DefaultCiphersInsufficient =
+            LoggerMessage.Define(
+                logLevel: LogLevel.Information,
```

Comment:
What is the right level for this? Debug? Information? Warning?

(I don't know, I'm posing the question to make sure it has been considered 😄 )

Reply:
I vote warning since this is only logged once at startup and can be silenced by configuring the endpoint to be HTTP/1.x-only.

Reply:
I did have it as a warning initially, then @Tratcher asked for Information/Debug: https://github.com/dotnet/aspnetcore/pull/22859#discussion_r439177620.

Reply:
Ping

Reply:
Warning didn't seem appropriate since this happens with the default kestrel configuration. You didn't do anything wrong, or even opt-into HTTP/2, and the site will probably work fine without HTTP/2.

Reply:
Fair enough. Information is fine then.

Reply:
Put another way, what are you warning them to do? Change the value to HTTP/1? We did that for them.